python/pyspark/ml/util.py

```diff
@@ -89,6 +89,8 @@ def __init__(self, instance):
 
     def save(self, path):
         """Save the ML instance to the input path."""
+        if not isinstance(path, basestring):
```

Comment:
To make sure this check happens, even if users create a writer and then call save.